website/docs/ref/react.doc.js

```diff
@@ -197,6 +197,28 @@ function renderButton(container: HTMLElement, visited?: boolean) {
 
   Any function that returns a React element can be used as a component class in
   a JSX expression.
+
+*/
+const AsyncGreeter = loadAsync(Greeter, new Promise((resolve, reject) => {
```

Comment:
Yeah, I was trying to come up with a common HOC example. I'll think of a better example.

Also, the types are a little weird when trying to also support stateless functions. I'm confused if I should focus on simplicity or resilience.